upgrader.go

```diff
@@ -56,28 +55,26 @@ func WithUpgradeTimeout(t time.Duration) Option {
 	}
 }
 
+// WithLogger configures the logger to use for tableroll operations.
+// By default, nothing will be logged.
 func WithLogger(l log15.Logger) Option {
 	return func(u *Upgrader) {
 		u.l = l
 	}
 }
 
+// New constructs a tableroll upgrader.
+// The first argument is a directory. All processes in an upgrade chain must
+// use the same coordination directory. The provided directory must exist and
+// be writeable by the process using tableroll.
+// Canonically, this directory is `/run/${program}/tableroll/`.
+// Any number of options to configure tableroll may also be provided.
 func New(coordinationDir string, opts ...Option) (upg *Upgrader, err error) {
-	stdEnvMu.Lock()
-	defer stdEnvMu.Unlock()
-	if stdEnvUpgrader != nil {
-		return nil, errors.New("tableroll: only a single Upgrader allowed")
-	}
-
-	upg, err = newUpgrader(coordinationDir, opts...)
-	// Store a reference to upg in a private global variable, to prevent
-	// it from being GC'ed and exitFd being closed prematurely.
-	stdEnvUpgrader = upg
-	return
+	return newUpgrader(realOS{}, coordinationDir, opts...)
 }
 
-func newUpgrader(coordinationDir string, opts ...Option) (*Upgrader, error) {
-	upgradeListener, err := listenSock(coordinationDir)
+func newUpgrader(os osIface, coordinationDir string, opts ...Option) (upg *Upgrader, err error) {
```

Comment:
Why the named returns? It doesn't seem strictly necessary, and the types already have useful names.

Reply:
leftover from tableflip, I'll nix them, thanks for spotting them.